bentoml/adapters/base_input.py

```diff
@@ -45,13 +45,14 @@ class BaseInputAdapter:
 
     BATCH_MODE_SUPPORTED = False
 
-    def __init__(self, output_adapter=None, **base_config):
+    def __init__(self, output_adapter=None, example_input=None, **base_config):
```

Comment:
consider renaming `example_input` to `http_input_example`?  the InputAdapter interface also handles input from CLI arguments, AWS Lambda event (and gRPC request, etc), it might be confusing to the user what example this is for